src/System.Private.Uri/tests/FunctionalTests/IriTest.cs

```diff
@@ -184,7 +184,7 @@ public void Iri_IncompleteEscapedCharacter()
         [Fact]
         public void Iri_ReservedCharacters()
         {
-            string input = "/?#??#%[]";
+            string input = "?/#??#%[]";
```

Comment:
This would actually cause an IndexOutOfRange exception to be thrown. I investigated it further now, and the cause was newHost being an empty string instead of null.
Reverted the IriTest and fixed here: https://github.com/tmds/corefx/commit/c13d66a84ed82401e6fd21e9d46b9050ca0e4503.